From 28f1b0cc2200cc33387fd96ade802f476dadb60e Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Thu, 15 Nov 2018 20:56:28 -0700
Subject: [PATCH 07/11] msm8974-common: sepolicy: Resolve hal_sensors_default
 denials

* avc: denied { read } for name="devices" dev="sysfs" ino=7783
  scontext=u:r:hal_sensors_default:s0 tcontext=u:object_r:sysfs:s0
  tclass=dir permissive=0
* avc: denied { open } for name="devices" dev="sysfs" ino=7783
  scontext=u:r:hal_sensors_default:s0 tcontext=u:object_r:sysfs:s0
  tclass=dir permissive=0
* avc: denied { read } for name="name" dev="sysfs" ino=26468i
  scontext=u:r:hal_sensors_default:s0 tcontext=u:object_r:sysfs:s0
  tclass=file permissive=0
* avc: denied { read } for name="iio:device1" dev="tmpfs" ino=7276
  scontext=u:r:hal_sensors_default:s0
  tcontext=u:object_r:iio_device:s0 tclass=chr_file permissive=0
* avc: denied { open } for name="iio:device0" dev="tmpfs" ino=7275
  scontext=u:r:hal_sensors_default:s0
  tcontext=u:object_r:iio_device:s0 tclass=chr_file permissive=0
* avc: denied { write } for name="length" dev="sysfs" ino=26482
  scontext=u:r:hal_sensors_default:s0 tcontext=u:object_r:sysfs_iio:s0
  tclass=file permissive=0
* avc: denied { read } for name="iio:device1" dev="sysfs" ino=26489
  scontext=u:r:hal_sensors_default:s0 tcontext=u:object_r:sysfs_iio:s0
  tclass=lnk_file permissive=0
* avc: denied { search } for name="/" dev="mmcblk0p12" ino=2
  scontext=u:r:hal_sensors_default:s0 tcontext=u:object_r:efs_file:s0
  tclass=dir permissive=0
* avc: denied { read } for name="gyro_cal_data" dev="mmcblk0p12"
  ino=41 scontext=u:r:hal_sensors_default:s0
  tcontext=u:object_r:efs_file:s0 tclass=file permissive=0
* avc: denied { search } for name="input" dev="sysfs" ino=13030
  scontext=u:r:hal_sensors_default:s0
  tcontext=u:object_r:sysfs_input:s0 tclass=dir permissive=0
* avc: denied { read } for name="input6" dev="sysfs" ino=26725
  scontext=u:r:hal_sensors_default:s0
  tcontext=u:object_r:sysfs_input:s0 tclass=dir permissive=0
* avc: denied { read } for name="device" dev="sysfs" ino=26717
  scontext=u:r:hal_sensors_default:s0
  tcontext=u:object_r:sysfs_input:s0 tclass=lnk_file permissive=0
* avc: denied { read write } for name="poll_delay" dev="sysfs"
  ino=26946 scontext=u:r:hal_sensors_default:s0
  tcontext=u:object_r:sysfs_input:s0 tclass=file permissive=0
* avc: denied { search } for name="lcd" dev="sysfs" ino=12695
  scontext=u:r:hal_sensors_default:s0
  tcontext=u:object_r:sysfs_graphics:s0 tclass=dir permissive=0
* avc: denied { read } for name="window_type" dev="sysfs" ino=12710
  scontext=u:r:hal_sensors_default:s0
  tcontext=u:object_r:sysfs_graphics:s0 tclass=file permissive=0

Change-Id: Iad7e41e5e250eb1511d5838bd42b2b07843d220b
---
 sepolicy/common/hal_sensors_default.te | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)
 create mode 100644 sepolicy/common/hal_sensors_default.te

diff --git a/sepolicy/common/hal_sensors_default.te b/sepolicy/common/hal_sensors_default.te
new file mode 100644
index 0000000..c737b8a
--- /dev/null
+++ b/sepolicy/common/hal_sensors_default.te
@@ -0,0 +1,21 @@
+r_dir_file(hal_sensors_default, efs_file)
+
+allow hal_sensors_default iio_device:chr_file r_file_perms;
+
+allow hal_sensors_default sysfs_graphics:dir search;
+allow hal_sensors_default sysfs_graphics:file r_file_perms;
+
+allow hal_sensors_default {
+    sysfs_iio
+    sysfs_input
+}:dir r_dir_perms;
+
+allow hal_sensors_default {
+    sysfs_iio
+    sysfs_input
+}:file rw_file_perms;
+
+allow hal_sensors_default {
+    sysfs_iio
+    sysfs_input
+}:lnk_file read;
-- 
2.17.1

